### PR TITLE
Bandit parameters compatibility with check in domain

### PR DESCRIPTION
**Pull request by ClaireCp**
*Issue #, if available:*

*Description of changes:*
The ParameterSpace check_points_in_domain is currently not compatible with bandit parameters because the call to "len(param.model_parameters)" on L127 of the parameter_space.py source will return 1 for a bandit parameter. 
By adding the property model_parameters() to the bandit implementation following the same logic of categorical parameters, we ensure the compatibility of check_points_in_domain with bandit parameters.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/emukit/core/test_parameter_space.py [hunk 1]
```diff
@@ -38,6 +38,17 @@ def test_check_in_domain(space_2d, space_3d_mixed):
     assert np.array_equal(in_domain, np.array([False, True]))
 
 
+def test_check_in_domain_with_bandit_parameter():
```
Comment (apaleyes) [inline tests/emukit/core/test_parameter_space.py:41]: Can you also add an isolated unit test, for a domain with bandit-only parameters?

**Review by apaleyes — COMMENTED** [review]
Thanks for your PR! Just a tiny comment, shouldn't take a long time to address.

**Review by apaleyes — APPROVED** [review]
Thanks, looks good!

File: tests/emukit/core/test_parameter_space.py (hunk 1 above)
Comment (ClaireCp) [reply tests/emukit/core/test_parameter_space.py:55]: Sorry for the delay!

**Comment by apaleyes** [comment]
Actually, hold on a sec. Can you have a look at the PR diff and confirm it is complete? I don't see any changes there from your first commit "fixed context manager for multiple categorical variables in context ". Where did they go? Are they still needed?

**Comment by ClaireCp** [comment]
Hi apaleyes, the changes from "fixed context manager for multiple categorical variables in context" have already been merged into master in a previous PR; I must have created this current branch on an old local version of emukit. I can create a new branch on top off the new version of master, readd the changes for bandit parameters and cut a new PR. That would take just a second and you'd have a clean git history. Let me know what you'd prefer.

**Comment by apaleyes** [comment]
No need, just checking. I'll squash this before merging anyway. Thanks for confirming promptly!

**Pull request merged**